### ISSUE TITLE
Use babel-polyfill instead of transform-runtime for client bundle

### ISSUE THREAD

**Issue by tizmagik** [NYTimes/kyt#227]
The babel transform-runtime plugin is meant more for libs and tools, while the polyfill is meant for production applications. Can we remove the plugin and instead use the polyfill for the client bundle?

https://babeljs.io/docs/usage/polyfill/

> If you are looking for something that won't modify globals to be used in a tool/library, checkout the transform-runtime plugin. This means you won't be able to use the instance methods mentioned above like Array.includes.

**Comment by dting** [comment]
@tizmagik I can make this PR if it is still needed. 

I just wanted to make sure I have all my bases covered:

* Uninstall and remove `babel-plugin-transform-runtime` dependency.
* Remove the `plugin` entry in the `.babelrc` file.
* Install and add `babel-polyfill` dependency. 
* Add `'babel-polyfill'` as the first element to the `main` array in `entry: { main: [] }` in `webpack.dev.client.js` and `webpack.prod.client.js` files.

I did the above and the polyfill code is being included in the compiled client bundles when I make a setup a new project and run `dev` and `build`. Do i need to add a test for this?

**Comment by tizmagik** [comment]
Thanks @dting ! I'm not sure if `babel-plugin-transform-runtime` is still necessary for the server-side, but I would think not, so yes, I think what you've outlined makes sense. We'll just need to test and ensure it all still runs under node 6 without the plugin.

Two additional things: 
 - Yes, a test would be good to ensure that the polyfill is correctly added as the first entry in main.
 - A note in the documentation would also be helpful to inform users that the polyfill is included for them.

**Comment by dting** [comment]
Thanks. I'll take a look at writing those tests and adding the note in the docs.

**Comment by dting** [comment]
@tizmagik I did some manual testing and it seems that switching to having the polyfill only in the client webpack config causes some issues for certain situations. 

The example that I tested was using the universal react starter kyt and using a generator function (`function *()`). When using `babel-plugin-transform-runtime` or having `babel-polyfill` in the server entry array, the code will run, however, if the server is not polyfilled and a generator is used things break down.

Any thoughts about including the polyfill in the server webpack config?
